autofaiss/external/build.py

```diff
@@ -189,7 +136,15 @@ def build_index(
             )
 
             # Extract training vectors
-            train_vectors = next(read_embeddings_local(embeddings_path, batch_size=train_size, verbose=True))
+            train_vectors = next(
```

Comment:
need to check that we are releasing the ram of that file here, I think it's not the case due to the fact it's an iterator

Reply:
maybe the fact we lose the reference to the iterator triggers the release

Reply:
Calling next again deletes the previous one

```python
from dataclasses import dataclass

@dataclass
class WithDestructor(object):
    i: int
    def __del__(self):
        print(f"delete {self.i}")


def my_destructors(n):
    for i in range(n):
        yield WithDestructor(i)


n = 3
i = 0
destructors_iter = my_destructors(n)

while i < n:
    print(f'i={i}')
    a = next(destructors_iter)
    print(a)
    i += 1
````

Outputs
```
i=0
WithDestructor(i=0)
i=1
delete 0
WithDestructor(i=1)
i=2
delete 1
WithDestructor(i=2)
delete 2
```

Reply:
ok sounds good